guacamole/src/main/frontend/src/app/client/types/ManagedFilesystem.js

```diff
@@ -147,13 +147,26 @@ angular.module('client').factory('ManagedFilesystem', ['$rootScope', '$injector'
                         if (mimetypes[name] === Guacamole.Object.STREAM_INDEX_MIMETYPE)
                             type = ManagedFilesystem.File.Type.DIRECTORY;
 
+                        // try deserialization fileJSON String to fileObject
+                        var size = 0;
+                        var permission = null;
+                        try {
+                            var fileObj = JSON.parse(mimetypes[name])
+                            size = fileObj.size || 0;
+                            if (fileObj.perm) {
+                                permission = ManagedFilesystem.permissionTranslate(fileObj.perm);
+                            }
+                        } catch (e) {
+                            return false;
```

Comment:
Why? `$evalAsync()` is not documented as doing anything with the return value of the provided callback.